front/src/applications/rollingStockEditor/components/OnboardSystemEquipmentForm.tsx

```diff
@@ -28,16 +30,19 @@ const RollingStockEditorOnboardSystemEquipmentForm = ({
 
   const updateSigSystemsList = (sigSystem: string) => (e: React.ChangeEvent<HTMLInputElement>) => {
     const newList = e.target.checked
-      ? [...rsSignalingSystemsList, sigSystem]
-      : rsSignalingSystemsList.filter((v) => v !== sigSystem);
+      ? [...rollingStockValues.supportedSignalingSystems, sigSystem]
+      : rollingStockValues.supportedSignalingSystems.filter((v) => v !== sigSystem);
     setRollingStockValues((prevRollingStockValues) => ({
       ...prevRollingStockValues,
       supportedSignalingSystems: newList,
     }));
   };
 
   const signalingSystemCheckboxes = sigSystemProperty.enum!.map((sigSystem, index) => {
-    const checked = rsSignalingSystemsList.includes(sigSystem);
+    const checked = [
+      ...rollingStockValues.supportedSignalingSystems,
+      ...(rollingStockValues?.etcsBrakeParams ? [ETCS_LEVEL2_SIGNALING_SYSTEM] : []),
```

Comment:
Not sure: I don't get why using the optional chaining (`?.`) here and not above
```suggestion
      ...(rollingStockValues.etcsBrakeParams ? [ETCS_LEVEL2_SIGNALING_SYSTEM] : []),
```